docs/migration.md

```diff
@@ -1,24 +1,256 @@
+### Create OSIAM's Home Directory
+
+OSIAM now loads assets and configuration from a folder in the filesystem.
+Create a home directory on the command line now:
+
+```sh
+java -jar osiam.war initHome --osiam.home=/var/lib/osiam
+```
+
+You can, of course, choose any other directory instead of `/var/lib/osiam`.
+Make sure that the user, who runs the command, can create or write the directory.
+
+If you have not changed any files under `/etc/osiam` (except for the configuration file),
+there's nothing more to do.
+If you have changed any files under `/etc/osiam` (except for the configuration file),
+you have to manually copy your existing configuration to the new home directory:
+
+```sh
+rm -rf /var/lib/osiam/i18n/*
+cp -r /etc/osiam/auth-server/i18n/* /var/lib/osiam/i18n/
+
+rm -rf /var/lib/osiam/css/*
+cp -r /etc/osiam/auth-server/resources/css/* /var/lib/osiam/css/
+
+rm -rf /var/lib/osiam/js/*
+cp -r /etc/osiam/auth-server/resources/js/* /var/lib/osiam/js/
+
+rm -rf /var/lib/osiam/templates/*
+cp -r /etc/osiam/auth-server/templates/* /var/lib/osiam/templates/
+```
+
+### Create a new Configuration File
+
+The new configuration file can be found under `$OSIAM_HOME/config/osiam.yaml`.
+Read it carefully, as this file also contains the explanation of each setting.
+Open the old configuration files and compare the settings to the new one.
+First set the the connection properties to the new database under `osiam.db`.
+Merge the connection pool properties:
+
+- `org.osiam.*-server.db.maximum-pool-size` becomes `osiam.db.maximum-pool-size`
+- `org.osiam.*-server.db.connection-timeout-ms` becomes `osiam.db.connection-timeout-ms`
+
+Migrate the LDAP configuration:
+
+```
+org.osiam.auth-server.ldap.enabled=true
+org.osiam.auth-server.ldap.server.url=ldap://ldap:389/dc=osiam,dc=org
+org.osiam.auth-server.ldap.dn.patterns=uid={0},ou=people
+org.osiam.auth-server.ldap.sync-user-data=true
+org.osiam.auth-server.ldap.mapping=userName:uid\
+,email:mail\
+,name.familyName:sn\
+,name.givenName:givenName\
+,displayName:displayName\
+```
+
+becomes
+
+```yaml
+osiam:
+
+  [...]
+
+  ldap:
+    enabled: true
+    server-url: ldap://ldap:389/dc=osiam,dc=org
+    dn-patterns:
+      - uid={0},ou=people
+    sync-user-data: true
+    user-mapping:
+      userName: uid
+      email: mail
+      name.familyName: sn
+      name.givenName: givenName
+      displayName: displayName
+``
+
+Migrate the temporary lock configuration:
+
+```
+org.osiam.auth-server.tempLock.count=10
+org.osiam.auth-server.tempLock.timeout=30
+```
+
+becomes
+
+```yaml
+osiam:
+
+  [...]
+
+  tempLock:
+    count: 10
+    timeout: 30
+```
+
+The following settings can be ignored, as they have been removed:
+
+- `org.osiam.*-server.home`
+- `org.osiam.*-server.connector.max-connections`
+- `org.osiam.*-server.connector.read-timeout-ms`
+- `org.osiam.*-server.connector.connect-timeout-ms`
+
+### Choose a Way of Deployment
+
+You have probably deployed OSIAM in a Tomcat Servlet container, before.
```

Comment:
drop the "have": You probably deployed OSIAM in a Tomcat servlet container before.